packages/core/src/config/config-resolvers.ts

```diff
@@ -73,43 +73,63 @@ export async function resolveConfigFileAndRefs({
   return { document, resolvedRefMap };
 }
 
+export type ConfigOptions = {
+  rawConfig?: RawUniversalConfig;
+  configPath?: string;
+  externalRefResolver?: BaseResolver;
+  customExtends?: string[];
+};
+
 export async function resolveConfig({
   rawConfig,
   configPath,
   externalRefResolver,
-}: {
-  rawConfig: RawConfig;
-  configPath?: string;
-  externalRefResolver?: BaseResolver;
-}): Promise<Config> {
-  if (rawConfig.styleguide?.extends?.some(isNotString)) {
-    throw new Error(
-      `Error configuration format not detected in extends value must contain strings`
-    );
+  customExtends,
+}: ConfigOptions): Promise<Config> {
+  const config = rawConfig === undefined ? { extends: ['recommended'] } : { ...rawConfig };
+  if (customExtends !== undefined) {
+    config.extends = customExtends;
+  }
+  if (config?.extends?.some(isNotString)) {
+    throw new Error(`Configuration format not detected in extends: values must be strings.`);
   }
 
-  const resolver = externalRefResolver ?? new BaseResolver(getResolveConfig(rawConfig.resolve));
+  const resolver = externalRefResolver ?? new BaseResolver(getResolveConfig(config.resolve));
 
   const apis = await resolveApis({
-    rawConfig,
+    rawConfig: config,
     configPath,
     resolver,
   });
 
-  const styleguide = await resolveStyleguideConfig({
-    styleguideConfig: rawConfig.styleguide,
+  const rootGovernanceConfig = await resolveGovernanceConfig({
+    rootOrApiRawConfig: config,
     configPath,
     resolver,
   });
 
-  return new Config(
-    {
-      ...rawConfig,
-      apis,
-      styleguide,
+  const { plugins: _plugins, extends: _extends, apis: _apis, ...rest } = config;
+  const resolvedConfig: ResolvedConfig = {
+    ...rest,
+    ...rootGovernanceConfig,
+    apis,
+  };
+
+  return {
```

Comment:
Returning config as an object instead of a class instance. It was somewhat confusing that we had that class and some people used it for instantiating the config (which might work incorrectly).

Reply:
This is a valid point 🤔

Reply:
but maybe we should just not export this class anymore directly.
Only loadConfig and crateConfig will be exported.